src/plugin-import-lvllvl.js

```diff
@@ -0,0 +1,265 @@
+/**
+🔠
+@file import lvllvl
+@summary Import text-mode art designed with lvllvl
+@license MIT
+@author Jean-Sébastien Monzani (base code by CrocMiam)
+@version 1.0
+
+
+@description
+Import JSON files exported from lvllvl https://lvllvl.com/
+
+
+HOW TO USE:
+1. In lvllvl:
+1.1 Make sure your palette has max 7 colors, indexed from 0
+1.2 Either design your rooms as a single long vertical image, or as frames in the animation timeline (bottom of the screen). For instance, a room is by default 16x16, and you could create a 16x32 image for two rooms or a 16x16 image with two animation frames
+1.3 Export your tileset to PNG (Tiles > Save Tile Set)
+1.4 Export your JSON as an image (Export > Export JSON and use the Native format. Tile Set Data isn't used so you can uncheck this)
+2. Import plugin in bipsi
+3. Go to edit mode, you'll find new buttons on the bottom right. First import your tilset, then the JSON image
+
+*/
+
+//! CODE_EDITOR
+//! CODE_EDITOR
+function U32ColorToRGBA(n) {
+    const u32a = new Uint32Array([n]);
+    const u8a = new Uint8ClampedArray(u32a.buffer);
+    return Array.from(u8a);
+}
+function autoCloseToggledWindow(windowElement, toggle) {
+    window.addEventListener("click", (event) => {
+        const target = event.target;
+        const ignore = windowElement.hidden ||
+            !event.isTrusted ||
+            windowElement.contains(target) ||
+            toggle.inputs.includes(target);
+        if (ignore)
+            return;
+        toggle.checked = false;
+    });
+}
+
+const defaultOptions = {
+    //roomStart: 0,
+};
+async function importMap(options) { //JSM
+    const fullOptions = { ...defaultOptions, ...options };
+    const [file] = await maker.pickFiles("application/json");
+    const mapdata = await maker.textFromFile(file);    
+    if (!file)
+        return;
+    // Create bipsi rooms & tiles
+    EDITOR.stateManager.makeChange(async (data) => {
+        const jsondata = JSON.parse(mapdata);
+        //import colors
+        let i = 1; //color 0 is reserved (transparent)
+        for (color of jsondata["colorPalette"]["data"]) {
```

Comment:
when i tested this plugin with the provided sample, it broke on this line bc `jsondata` had no `colorPalette`, i'm not familiar with the format but i'm guessing it needs a nullish check before it tries to loop through here?

Reply:
Yeah that's because I didn't provide a json file in lvllvl format, just one for Tiled.